### DISCUSSION TITLE
Questions about config replication and auto failover of EMQX cluster

### DISCUSSION THREAD

**Discussion by alanko0511** [emqx/emqx#6159] [Q&A] [answered]
Hi!

I have used single-node EMQX, and it works great, so I would like to know more about setting up an EMQX cluster. Currently, I only want to set up using the "manual" method instead of k8s or etcd, and all connections will go through GCP Cloud Load Balancer.

1. The cluster will require HTTP authentication for every MQTT connection and other setting changes, such as rate limit and rule engine. So do I need to modify the configs manually for every node? Or will it sync automatically to other unmodified nodes?

2. If a node (node A) suddenly disconnects from the cluster (e.g. network issue or crash), while all clients that subscribe to a topic, let say "/notification" is connected to that node. What would happen when a message published to "/notification" from another node with QoS 1? Is the message still available after a failover from the load balancer then the clients are connected to a healthy node?

Sorry a bit wordy, but I want to know more about how EMQX cluster works.

**Comment by k32** [comment, marked as answer]
Hello,
1. In the current version (4.*) some settings are replicated across the cluster (e.g. rule engine rules), while others are specific for the node. (Ones that are configured using config files). In 5.0 it will be more consistent.
2. In the event of the netsplit, the clients connected to the different partitions won't see each other, and the messages won't be redelivered once the partition is healed.
